src/bosh-director/lib/bosh/director/starter.rb

```diff
@@ -1,33 +1,63 @@
 module Bosh::Director
   class Starter
     class << self
-      def start(
-        instance:,
-        agent_client:,
-        update_config:,
-        is_canary: false,
-        wait_for_running: true,
-        logger: Config.logger
-      )
-        logger.info("Running pre-start for #{instance}")
-        agent_client.run_script('pre-start', {})
-
-        logger.info("Starting instance #{instance}")
-        agent_client.start
+      def start(args = {})
+        instance, agent_client, update_config = parse_required(args)
+        wait_for_running, task, logger, is_canary = parse_optional(args)
+        run_pre_start(instance, agent_client, task, logger)
+        start_jobs(instance, agent_client, task, logger)
 
         return unless update_config && wait_for_running
 
+        min_watch_time, max_watch_time = min_max_watch_time(is_canary,
+                                                            update_config)
+
+        wait_until_running(instance, agent_client, min_watch_time,
+                           max_watch_time, logger)
+        run_post_start(instance, agent_client, task, logger)
+      end
+
+      private
+
+      def min_max_watch_time(is_canary, update_config)
         min_watch_time = is_canary ? update_config.min_canary_watch_time : update_config.min_update_watch_time
         max_watch_time = is_canary ? update_config.max_canary_watch_time : update_config.max_update_watch_time
+        [min_watch_time, max_watch_time]
+      end
+
+      def parse_required(args)
+        instance = args.fetch(:instance)
+        agent_client = args.fetch(:agent_client)
+        update_config = args.fetch(:update_config)
+        [instance, agent_client, update_config]
+      end
+
+      def parse_optional(args)
+        wait_for_running = args.fetch(:wait_for_running, true)
+        task = args.fetch(:task, nil)
```

Comment:
I think there should always be a task involved. I think it's up to you if you want to take on the overhead of trying to get the task through all the other calls to this method though but recreates and restarts won't have these logs when going through [this code path](https://github.com/cloudfoundry/bosh/blob/418b288d8b9a7fec781396efbe050e2c0054789b/src/bosh-director/lib/bosh/director/jobs/start_instance.rb#L25). It would be totally fair to say that it's a separate story.

At the very least you can remove all the `&` by using the pre-existing [NullStage](https://github.com/cloudfoundry/bosh/blob/41b673796fd7c42bb6fdb7c2c9688368d2efe448/src/bosh-director/lib/bosh/director/event_log.rb#L124)